apps/sms/views/conversation/test/unit/startup_test.js

```diff
@@ -53,62 +52,87 @@ suite('ConversationView Startup', function() {
-    MessageManager.getThreads.lastCall.args[0].done();
-    Promise.resolve().then(() => {
-      sinon.assert.calledWithExactly(
-        Navigation.toDefaultPanel,
-        Utils.params('#param1=1&param2=2')
-      );
-    }).then(done, done);
+    this.sinon.stub(Navigation, 'isDefaultPanel');
   });
 
-  test('starts non-critical initializations only once view is visually ready',
-  function() {
-    // Lazy loading is not started yet and headers are not processed either.
-    sinon.assert.notCalled(LazyLoader.load);
-    assert.isTrue(
-      document.querySelectorAll('gaia-header[no-font-fit]').length > 0
-    );
-
-    ConversationView.once.withArgs('visually-loaded').yield();
-
-    sinon.assert.calledOnce(LazyLoader.load);
-    assert.isTrue(
-      document.querySelectorAll('gaia-header[no-font-fit]').length === 0
-    );
+  suite('In default panel,', function() {
+    setup(function() {
+      Navigation.isDefaultPanel.returns(true);
+
+      Startup.init();
+    });
+
+    test('correctly initializes dependencies', function(done) {
+      sinon.assert.calledOnce(MessageManager.init);
+      sinon.assert.calledOnce(ConversationView.init);
+
+      // Workaround to deal with temporary shim code.
+      MessageManager.getThreads.lastCall.args[0].done();
+      Promise.resolve().then(
+        () => sinon.assert.calledOnce(Navigation.init)
+      ).then(done, done);
+    });
+
+    test('starts non-critical initializations only once view is visually ready',
+    function() {
+      // Lazy loading is not started yet and headers are not processed either.
+      sinon.assert.notCalled(LazyLoader.load);
+
+      ConversationView.once.withArgs('visually-loaded').yield();
+
+      sinon.assert.calledOnce(LazyLoader.load);
+    });
+
+    test('correctly initializes lazy dependencies', function(done) {
+      ConversationView.once.withArgs('visually-loaded').yield();
+
+      LazyLoader.load.lastCall.returnValue.then(() => {
+        sinon.assert.calledOnce(LocalizationHelper.init);
+        sinon.assert.calledOnce(TimeHeaders.init);
+        sinon.assert.calledOnce(Settings.init);
+        sinon.assert.calledOnce(Navigation.setReady);
+        sinon.assert.calledOnce(Information.initDefaultViews);
+        sinon.assert.calledOnce(InterInstanceEventDispatcher.connect);
+      }).then(done, done);
+    });
   });
 
-  test('correctly initializes lazy dependencies', function(done) {
-    ConversationView.once.withArgs('visually-loaded').yield();
-
-    LazyLoader.load.lastCall.returnValue.then(() => {
-      sinon.assert.calledOnce(LocalizationHelper.init);
-      sinon.assert.calledOnce(TimeHeaders.init);
-      sinon.assert.calledOnce(Settings.init);
-      sinon.assert.calledOnce(Navigation.setReady);
-      sinon.assert.calledOnce(Information.initDefaultViews);
-      sinon.assert.calledOnce(InterInstanceEventDispatcher.connect);
-    }).then(done, done);
+  suite('In a subpanel,', function() {
```

Comment:
__question:__ same question here - how can we load ```views/conversation/js/startup.js``` and be in a subpanel (IIRC it's loaded only in ```views/conversation/index.html``` that can only be loaded as a separate document)?